### PR TITLE
feat: `avm/res/databricks/access-connector`

### PR DESCRIPTION
**Pull request by segraef**
[avm.res.databricks.access-connector](https://github.com/segraef/bicep-registry-modules/actions/workflows/avm.res.databricks.access-connector.yml)

### CODE REVIEW

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
Requires uncommenting in the `/.github/CODEOWNERS` file

File: avm/res/databricks/access-connector/main.bicep [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+@description('Optional. Array of role assignments to create.')
+param roleAssignments roleAssignmentType
+
+@description('Optional. The lock settings of the service.')
+param lock lockType
+
+@description('Optional. The managed identity definition for this resource.')
+param managedIdentities managedIdentitiesType
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
```
Comment (AlexanderSehr) [inline avm/res/databricks/access-connector/main.bicep:23]:
```suggestion
@description('Optional. Enable/Disable usage telemetry for module.')
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -34,7 +34,7 @@
 #/avm/res/container-instance/container-group/ @Azure/avm-res-containerinstance-containergroup-module-owners-bicep @Azure/avm-core-team-technical-bicep
 #/avm/res/container-registry/registry/ @Azure/avm-res-containerregistry-registry-module-owners-bicep @Azure/avm-core-team-technical-bicep
 #/avm/res/container-service/managed-cluster/ @Azure/avm-res-containerservice-managedcluster-module-owners-bicep @Azure/avm-core-team-technical-bicep
-#/avm/res/databricks/access-connector/ @Azure/avm-res-databricks-accessconnector-module-owners-bicep @Azure/avm-core-team-technical-bicep
+/avm/res/databricks/access-connector/ @Azure/avm-res-databricks-accessconnector-module-owners-bicep @Azure/avm-core-team-technical-bicep
```
Comment (AlexanderSehr) [inline .github/CODEOWNERS:37]: Team must be created

File: avm/res/databricks/access-connector/main.bicep [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+output name string = accessConnector.name
+
+@description('The resource ID of the deployed access connector.')
+output resourceId string = accessConnector.id
+
+@description('The resource group of the deployed access connector.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The principal ID of the system assigned identity.')
+output systemAssignedMIPrincipalId string = (managedIdentities.?systemAssigned ?? false) && contains(accessConnector.identity, 'principalId') ? accessConnector.identity.principalId : ''
```
Comment (AlexanderSehr) [inline avm/res/databricks/access-connector/main.bicep:100]:
```suggestion
output systemAssignedMIPrincipalId string = (managedIdentities.?systemAssigned ?? false) ? accessConnector.identity.?principalId : ''
```

File: .github/CODEOWNERS (hunk 2 above)
Comment (AlexanderSehr) [inline .github/CODEOWNERS:37]: Team must be created

File: avm/res/databricks/access-connector/main.bicep (hunk 3 above)
Comment (AlexanderSehr) [inline avm/res/databricks/access-connector/main.bicep:100]:
```suggestion
output systemAssignedMIPrincipalId string = (managedIdentities.?systemAssigned ?? false) ? accessConnector.identity.?principalId : ''
```

File: .github/CODEOWNERS (hunk 2 above)
Comment (eriqua) [reply .github/CODEOWNERS:37]: And added to parent team as per https://azure.github.io/Azure-Verified-Modules/specs/shared/#team-memberships

File: .github/workflows/avm.res.databricks.access-connector.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+concurrency:
+  group: ${{ github.workflow }}
+
+jobs:
+  ###########################
+  #   Initialize pipeline   #
+  ###########################
+  job_initialize_pipeline:
+    runs-on: ubuntu-20.04
```
Comment (eriqua) [inline .github/workflows/avm.res.databricks.access-connector.yml:47]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/CODEOWNERS (hunk 2 above)
Comment (segraef) [reply .github/CODEOWNERS:37]: @Azure/avm-res-databricks-accessconnector-module-owners-bicep
@Azure/avm-res-databricks-accessconnector-module-contributors-bicep

File: avm/res/databricks/access-connector/main.bicep [hunk 5]
```diff
@@ -97,7 +97,7 @@ output resourceId string = accessConnector.id
 output resourceGroupName string = resourceGroup().name
 
 @description('The principal ID of the system assigned identity.')
-output systemAssignedMIPrincipalId string = (managedIdentities.?systemAssigned ?? false) && contains(accessConnector.identity, 'principalId') ? accessConnector.identity.principalId : ''
+output systemAssignedMIPrincipalId string = (managedIdentities.?systemAssigned ?? false) ? accessConnector.identity.?principalId : ''
```
Comment (AlexanderSehr) [inline avm/res/databricks/access-connector/main.bicep:100]:
```suggestion
output systemAssignedMIPrincipalId string = accessConnector.?identity.?principalId ?? ''
```

File: .github/CODEOWNERS (hunk 2 above)
Comment (AlexanderSehr) [reply .github/CODEOWNERS:37]: Still shows errors. It might be missing the owner assignment? / it might still need approval.

Comment (segraef) [reply .github/CODEOWNERS:37]: Both teams exist and parent teams are assigned as well.

File: avm/res/databricks/access-connector/main.json
Comment (AlexanderSehr) [inline avm/res/databricks/access-connector/main.json:1]: May be outdated after recent outputs update

File: avm/res/databricks/access-connector/tests/e2e/waf-aligned/main.test.bicep [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    }
+    managedIdentities: {
+      systemAssigned: true
+    }
+    tags: {
+      'hidden-title': 'This is visible in the resource name'
+      Environment: 'Non-Prod'
+      Role: 'DeploymentValidation'
+    }
+    location: resourceGroup.location
```
Comment (AlexanderSehr) [inline avm/res/databricks/access-connector/tests/e2e/waf-aligned/main.test.bicep:56]:
```suggestion
    location: location
```

Comment (AlexanderSehr) [reply avm/res/databricks/access-connector/tests/e2e/waf-aligned/main.test.bicep:56]: Note: Requires an update of the readme

File: avm/res/databricks/access-connector/tests/e2e/max/main.test.bicep [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        principalId: nestedDependencies.outputs.managedIdentityPrincipalId
+        principalType: 'ServicePrincipal'
+      }
+    ]
+    tags: {
+      'hidden-title': 'This is visible in the resource name'
+      Environment: 'Non-Prod'
+      Role: 'DeploymentValidation'
+    }
+    location: resourceGroup.location
```
Comment (AlexanderSehr) [inline avm/res/databricks/access-connector/tests/e2e/max/main.test.bicep:85]:
```suggestion
    location: location
```

File: avm/res/databricks/access-connector/main.json
Comment (segraef) [reply avm/res/databricks/access-connector/main.json:1]: updated, running workflow again

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**